### PR TITLE
Scenario6

### PR DESCRIPTION
**Pull request by irina-pereiaslavskaia**
Close: #238

### CODE REVIEW

File: playbooks/files/rds/check_rds_filling_status.sh [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+     `--host ${pg_host} `
+     `--port ${pg_port} `
+     `--username ${pg_username} `
+     `--password ${pg_password} `
+     `--database ${pg_database} `
+     `--drivername ${drivername}
+    echo "rds filling script is restarted"
+else
+    echo "rds filling script finished"
+fi
```
Comment (anton-sidelnikov) [inline playbooks/files/rds/check_rds_filling_status.sh:30]: add newline

File: playbooks/files/rds/pg2/data.yaml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+---
+   record_count: 14027776
+   max_size_in_bytes: 10737418240
```
Comment (anton-sidelnikov) [inline playbooks/files/rds/pg2/data.yaml:3]: newline

File: playbooks/files/rds/sqla/data.yaml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+---
+   record_count: 1024000
+   symbol_count: 975
+   max_size_in_bytes: 10737418240
```
Comment (anton-sidelnikov) [inline playbooks/files/rds/sqla/data.yaml:4]: newline

File: playbooks/files/rds/check_rds_filling_status.sh (hunk 1 above)
Comment (irina-pereiaslavskaia) [reply playbooks/files/rds/check_rds_filling_status.sh]: Done

File: playbooks/files/rds/pg2/data.yaml (hunk 2 above)
Comment (irina-pereiaslavskaia) [reply playbooks/files/rds/pg2/data.yaml]: Done

File: playbooks/files/rds/sqla/data.yaml (hunk 3 above)
Comment (irina-pereiaslavskaia) [reply playbooks/files/rds/sqla/data.yaml]: Done

File: playbooks/files/rds/pg2/db_methods.py [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+    create_table_query = model_query.format(
+                                            sql.Identifier(schema_name),
+                                            sql.Identifier(table_name),
+                                            sql.SQL(', ').join(map(sql.Identifier, columns))
+                                           )
+    _execute_sql(create_table_query)
+    logging.info('Table was created (if it not existed before)')
+
+
+def generate_random_values_and_insert_into_table(schema_name: str, table_name: str, range_start: int, range_stop: int, *columns):
```
Comment (lego963) [inline playbooks/files/rds/pg2/db_methods.py:53]: `too_long_func_name`
what about `fill_table`?

File: playbooks/files/rds/pg2/db_methods.py [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+    with open(get_source()) as data_file:
+        data = yaml.safe_load(data_file)
+        n = data['record_count']
+        i = 1
+        schema_name = 'public'
+        table_name = str(uuid.uuid4())
+        while not is_database_fulfilled('entities', data['max_size_in_bytes']):
+            create_table(schema_name, table_name, 'content')
+            generate_random_values_and_insert_into_table(schema_name, table_name, i + (i - 1) * n, i * n, 'content')
+            i = i + 1
```
Comment (lego963) [inline playbooks/files/rds/pg2/db_methods.py:90]: i++

File: playbooks/scenario6_setup.yml [hunk 6]
```diff
@@ -0,0 +1,219 @@
...
+            extra_parameters: |
+              location / {
+                  proxy_set_header X-Forward-For $proxy_add_x_forwarded_for;
+                  proxy_set_header Host $http_host;
+                  proxy_http_version 1.1;
+                  proxy_pass         http://localhost:{{ server_local_port }};
+              }
+              location /telegraf {
+                  proxy_pass         http://localhost:{{ telegraf_local_port }};
+                  proxy_set_header   Host $host;
```
Comment (lego963) [inline playbooks/scenario6_setup.yml:180]: miss `}`

File: playbooks/files/rds/pg2/db_methods.py (hunk 5 above)
Comment (irina-pereiaslavskaia) [reply playbooks/files/rds/pg2/db_methods.py]: i+=1, i++ not used

File: playbooks/scenario6_setup.yml (hunk 6 above)
Comment (irina-pereiaslavskaia) [reply playbooks/scenario6_setup.yml:180]: Done

File: playbooks/files/rds/pg2/db_methods.py (hunk 4 above)
Comment (irina-pereiaslavskaia) [reply playbooks/files/rds/pg2/db_methods.py]: Change to fill_table_by_random_values

**Review by Polina-Gubina — APPROVED** [review]

File: playbooks/prepare_config.yml [hunk 7]
```diff
@@ -14,4 +14,4 @@
   - name: Copy clouds.yaml
     copy:
       src: "{{ clouds_yaml_file }}"
-      dest: "{{ clouds_yaml_file }}"
+      dest: "{{ clouds_yaml_file }}"
```
Comment (anton-sidelnikov) [inline playbooks/prepare_config.yml:17]: empty string)

File: playbooks/scenario6_destroy.yml [hunk 8]
```diff
@@ -0,0 +1,23 @@
+---
+- name: Stop Server
+  hosts: scn6_ecs
+  become: yes
+  roles:
+    - role: too_simple_server
+  vars:
+    server_state: absent
```
Comment (anton-sidelnikov) [inline playbooks/scenario6_destroy.yml:8]: you can skip this

File: playbooks/scenario6_destroy.yml [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+  hosts: test_host
+  environment:
+    OS_CLIENT_CONFIG_FILE: "{{ clouds_yaml_file }}"
+    TF_VAR_psql_password: "{{ psql_password }}"
+  vars:
+    terraform_base_dir: "terraform"
+    requirements: "requirements.txt"
+    infra_state: absent
+  roles:
+    - build_infrastructure
```
Comment (anton-sidelnikov) [inline playbooks/scenario6_destroy.yml:23]: empty string

Comment (irina-pereiaslavskaia) [reply playbooks/scenario6_destroy.yml]: Done

**Review by Polina-Gubina — APPROVED** [review]

File: scenarios/scenario6/infrastructure.tf [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+  value = module.postgresql.db_address
+}
+
+output "db_host" {
+  value = module.postgresql.db_host
+}
+
+output "db_port" {
+  value = module.postgresql.db_port
+}
```
Comment (anton-sidelnikov) [inline scenarios/scenario6/infrastructure.tf:57]: empty string

File: scenarios/scenario6/resources/net.tf [hunk 11]
```diff
@@ -0,0 +1,29 @@
+#need to combine with scenario2/resources and create module
```
Comment (anton-sidelnikov) [inline scenarios/scenario6/resources/net.tf:1]: remove it

File: scenarios/scenario6/resources/variables.tf [hunk 12]
```diff
@@ -0,0 +1,8 @@
+variable "prefix" {}
+variable "net_address" {}
+variable "region" {}
+variable "availability_zone" {}
+variable "scenario" {}
+variable "public_key" {default = ""}
+variable "ecs_flavor" {}
+variable "ecs_image" {}
```
Comment (anton-sidelnikov) [inline scenarios/scenario6/resources/variables.tf:8]: empty string

File: scenarios/scenario6/resources/vm.tf [hunk 13]
```diff
@@ -0,0 +1,84 @@
...
+resource "opentelekomcloud_compute_keypair_v2" "pair" {
+  name       = "${var.prefix}_kp"
+  public_key = var.public_key
+}
+
+#Elastic ip creation
+resource "opentelekomcloud_vpc_eip_v1" "scn_eip" {
+
+  bandwidth {
+    name       = "scn_limit"
```
Comment (anton-sidelnikov) [inline scenarios/scenario6/resources/vm.tf:11]: scn6_limit

File: scenarios/scenario6/resources/vm.tf [hunk 14]
```diff
@@ -0,0 +1,84 @@
...
+
+  publicip {
+    type = "5_bgp"
+  }
+}
+
+#Security group creation
+resource "opentelekomcloud_compute_secgroup_v2" "scn_public_secgroup" {
+  description = "Allow external connections to ssh, http, and https ports"
+  name        = "scn_public_secgroup_rds"
```
Comment (anton-sidelnikov) [inline scenarios/scenario6/resources/vm.tf:24]: scn6_public_secgroup_rds

File: scenarios/scenario6/resources/vm.tf [hunk 15]
```diff
@@ -0,0 +1,84 @@
...
+#assigning ip
+resource "opentelekomcloud_compute_floatingip_associate_v2" "assign_ip" {
+  floating_ip = opentelekomcloud_vpc_eip_v1.scn_eip.publicip[0].ip_address
+  instance_id = opentelekomcloud_compute_instance_v2.basic.id
+}
+
+#output value of elastic ip
+output "scn_eip" {
+  value = opentelekomcloud_vpc_eip_v1.scn_eip.publicip[0].ip_address
+}
```
Comment (anton-sidelnikov) [inline scenarios/scenario6/resources/vm.tf:84]: empty string

File: scenarios/scenario6/variables.tf [hunk 16]
```diff
@@ -0,0 +1,12 @@
...
+variable "addr_3_octets" {default = "192.168.0"}
+variable "region" {default = "eu-de"}
+variable "availability_zone" {default = "eu-de-03"}
+variable "ecs_flavor" {}
+variable "ecs_image" {}
+variable "scenario" {}
+variable "psql_version" {}
+variable "psql_port" {}
+variable "psql_password" {}
+variable "public_key" {}
```
Comment (anton-sidelnikov) [inline scenarios/scenario6/variables.tf:12]: empty string

File: scenarios/scenario6/resources/net.tf (hunk 11 above)
Comment (irina-pereiaslavskaia) [reply scenarios/scenario6/resources/net.tf]: Done

File: scenarios/scenario6/resources/variables.tf (hunk 12 above)
Comment (irina-pereiaslavskaia) [reply scenarios/scenario6/resources/variables.tf]: Done

File: scenarios/scenario6/resources/vm.tf (hunk 13 above)
Comment (irina-pereiaslavskaia) [reply scenarios/scenario6/resources/vm.tf]: Done

File: scenarios/scenario6/resources/vm.tf (hunk 14 above)
Comment (irina-pereiaslavskaia) [reply scenarios/scenario6/resources/vm.tf]: Done

File: scenarios/scenario6/resources/vm.tf (hunk 15 above)
Comment (irina-pereiaslavskaia) [reply scenarios/scenario6/resources/vm.tf]: Done

**Review by anton-sidelnikov — APPROVED** [review]

**Review by Polina-Gubina — APPROVED** [review]

**Pull request merged**